inc/sainexthop.h

```diff
@@ -229,6 +229,28 @@ typedef enum _sai_next_hop_attr_t
      */
     SAI_NEXT_HOP_ATTR_DECREMENT_TTL,
 
+    /**
+     * @brief Encap tunnel-id for MPLS PUSH
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_TUNNEL
+     * @allownull true
+     * @condition SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_MPLS
+     */
+    SAI_NEXT_HOP_ATTR_MPLS_ENCAP_TUNNEL_ID,
+
+    /**
+     * @brief Decap tunnel-id for MPLS PHP
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_TUNNEL
+     * @allownull true
+     * @condition SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_IP
+     */
+    SAI_NEXT_HOP_ATTR_MPLS_DECAP_TUNNEL_ID,
```

Comment:
is this behavior backward compatible? it looks now if the next hop is IP, then user must include this new attribute? it seems like a non-backward compatible behavior.

Reply:
For decapsulation (PHP) this new optional attribute is proposed. If provided, then tunnel modes can be used to identify TTL and QoS treatment for outgoing traffic. 

Next hop of type IP is used for PHP as per https://github.com/opencomputeproject/SAI/blob/master/doc/MPLS/SAI-Proposal-MPLS-ver4.docx

Reply:
I plan to move this to InSegEntry as per comment by with Haidong last week. This is to be used in conjunction with IP next hop to provide decapsulation behavior for PHP